cmd/createorupdate/createorupdate.go

```diff
@@ -263,6 +267,44 @@ func (s *server) reply(w http.ResponseWriter, req *http.Request) {
 	w.Write(res)
 }
 
+var conf config
+
+type config struct {
+	SubscriptionID   string `envconfig:"AZURE_SUBSCRIPTION_ID" required:"true"`
+	TenantID         string `envconfig:"AZURE_TENANT_ID" required:"true"`
+	ClientID         string `envconfig:"AZURE_CLIENT_ID" required:"true"`
+	ClientSecret     string `envconfig:"AZURE_CLIENT_SECRET" required:"true"`
+	AADClientID      string `envconfig:"AZURE_AAD_CLIENT_ID"`
+	Region           string `envconfig:"AZURE_REGION"`
+	DnsDomain        string `envconfig:"DNS_DOMAIN" required:"true"`
+	DnsResourceGroup string `envconfig:"DNS_RESOURCEGROUP" required:"true"`
+	ResourceGroup    string `envconfig:"RESOURCEGROUP" required:"true"`
+
+	NoGroupTags      bool   `envconfig:"NOGROUPTAGS"`
+	ResourceGroupTTL string `envconfig:"RESOURCEGROUP_TTL"`
+}
+
+func (c *config) init() error {
+	supportedRegions := []string{"eastus", "westeurope", "australiasoutheast"}
+	if c.Region == "" {
+		// Randomly assign a supported region
+		rand.Seed(time.Now().UTC().UnixNano())
+		c.Region = supportedRegions[rand.Intn(3)]
+		logrus.Infof("Using randomly selected region %q", c.Region)
```

Comment:
you could also just default to eastus

Reply:
The point of using random regions is to have our tests ensure we can deploy in every supported region. We could have more scoped tests that always deploy in a specific region but it's more unnecessary work at this point.